packages/grid/src/GridMetricCalculator.ts

```diff
@@ -836,9 +836,12 @@ export class GridMetricCalculator {
     visibleHeight: number = this.getVisibleHeight(state)
   ): VisibleIndex {
     const { model } = state;
-    const { rowCount, floatingBottomRowCount } = model;
+    const { rowCount, floatingTopRowCount, floatingBottomRowCount } = model;
```

Comment:
It purposely does not take into account the `floatingTopRowCount` here - this is because the `lastTop` is actually scrolled to underneath the floating rows. This is incorrect, you can check this by going to the Iris Grid example in style guide and scrolling to the bottom, you'll be missing ~4 rows.
`getLastLeft` should probably take the `floatingRightColumnCount` into account, and we just haven't noticed that issue because we never use floating right columns.

Can you also fix the comment up for this method? There's a duplicated column block right before this one.

Reply:
![image](https://user-images.githubusercontent.com/1576283/205112048-77cf2c73-24f3-472b-b3ce-85c982cb58d6.png)

Hmm, but if I add isStuckToBottom to that example without accounting for top floating rows it initializes like this though.? If I scroll up then down it's right again though. Not sure why though

Reply:
I don't really get why it wasn't working, and how I fixed it.... lastTop was different between the results of `getMetricState` and the value of `lastTop` in `this.metrics`, switched to just use the value of lastTop in metrics, and it seems to be correct now?

Reply:
Okay, the root cause was I was calling lastTop with only two params, instead of three. The default one in metrics is called with 3, with a different value for the last param then the default which accounts for the difference.

To make this less likely of an error to occur in future, I have chosen to make them all required params.

same with getLastLeft. Please check what I did to focusFilterBar call to make sure that makes sense.